File: multi-architecture-app/node-express/src/services/greetingService.js
const GreetingRepository = require("../repositories/greetingRepository");
const GreetingResponse = require("../models/greetingResponse");

class GreetingService {
  constructor(repository = new GreetingRepository()) {
    this.repository = repository;
  }

  getGreeting() {
    const message = this.repository.getGreeting();
    return new GreetingResponse(message);
  }
}

module.exports = GreetingService;

File: multi-architecture-app/python-fastapi/app/services/greeting_service.py
from app.models.greeting import GreetingResponse
from app.repositories.greeting_repository import GreetingRepository


class GreetingService:
    """Aplica regras de negócio sobre os dados do repositório."""

    def __init__(self, repository: GreetingRepository | None = None) -> None:
        self.repository = repository or GreetingRepository()

    def get_greeting(self) -> GreetingResponse:
        message = self.repository.get_greeting()
        return GreetingResponse(message=message)
